deploy/vagrant/scripts/tinkerbell.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# abort this script on errors
+set -euxo pipefail
+
+whoami
+
+cd /vagrant
+
+setup_docker() (
+	# steps from https://docs.docker.com/engine/install/ubuntu/
+	sudo apt-get install -y \
+	     apt-transport-https \
+	     ca-certificates \
+	     curl \
+	     gnupg-agent \
+	     software-properties-common
+
+	curl -fsSL https://download.docker.com/linux/ubuntu/gpg \
+		| sudo apt-key add -
+
+	local repo
+	repo=$(
+		printf "deb [arch=amd64] https://download.docker.com/linux/ubuntu %s stable" \
+		       "$(lsb_release -cs)"
+	    )
+	sudo add-apt-repository "$repo"
+
+	sudo apt-get update
+	sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+
+       	sudo usermod -aG docker "$USER"
+
+	newgrp
+)
+
+setup_docker_compose() (
+	# from https://docs.docker.com/compose/install/
+	sudo curl -L \
+	     "https://github.com/docker/compose/releases/download/1.26.0/docker-compose-$(uname -s)-$(uname -m)" \
+	     -o /usr/local/bin/docker-compose
+
+	sudo chmod +x /usr/local/bin/docker-compose
+)
+
+make_certs_writable() (
```

Comment:
can you please explain why are you using this form of declaring functions? please answer in https://github.com/tinkerbell/tink/issues/171 if you can :)